map/set-max-extent/src/main/java/com.esri.samples.set_max_extent/SetMaxExtentSample.java

```diff
@@ -0,0 +1,142 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.set_max_extent;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.control.ToggleGroup;
+import javafx.scene.layout.*;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.symbology.*;
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.BasemapStyle;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class SetMaxExtentSample extends Application {
+
+    private MapView mapView;
+
+
+    @Override
+    public void start(Stage stage) {
+
+        try {
+            // create stack pane and application scene
+            var stackPane = new StackPane();
+            var scene = new Scene(stackPane);
+
+            // set title, size, and add scene to stage
+            stage.setTitle("Set Max Extent Sample");
+            stage.setWidth(800);
+            stage.setHeight(700);
+            stage.setScene(scene);
+            stage.show();
+
+            // authentication with an API key or named user is required to access basemaps and other location services
+            String yourAPIKey = System.getProperty("apiKey");
+            ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+            // create a map with the streets basemap style
+            var map = new ArcGISMap(BasemapStyle.ARCGIS_STREETS);
+
+            // create a map view and set the map to it
+            mapView = new MapView();
+            mapView.setMap(map);
+
+            // create a max extent envelope with points on Colorado's northwest and southeast corners
+            var coloradoNW = new Point(-12139393.2109, 5012444.0468, SpatialReferences.getWebMercator());
+            var coloradoSE = new Point(-11359277.5124, 4438148.7816, SpatialReferences.getWebMercator());
+            var extentEnvelope = new Envelope(coloradoNW, coloradoSE);
+
+            // set the map's max extent to the max extent envelope
+            map.setMaxExtent(extentEnvelope);
+
+            // create a graphics overlay of the map's max extent and add it to the map view
+            var coloradoGraphicsOverlay = new GraphicsOverlay();
+            coloradoGraphicsOverlay.getGraphics().add(new Graphic(map.getMaxExtent()));
```

Comment:
it would be preferable if the new graphic was created from the envelope here: using map.getMaxExtent in a simple sample like this one is unlikely to cause any async issues, but a bigger app might (the setting of the map's max extent might not have completed before the new graphic is created). Let's use the envelope instead which has been instantiated on the same thread above and is safer:

```suggestion
            coloradoGraphicsOverlay.getGraphics().add(new Graphic(envelope));
```